packages/gatsby/src/schema/graphql-engine/bundle-webpack.ts

```diff
@@ -150,16 +152,71 @@ export async function createGraphqlEngineBundle(
   })
 
   return new Promise((resolve, reject) => {
-    compiler.run((err, stats) => {
-      compiler.close(closeErr => {
-        if (err) {
-          return reject(err)
+    compiler.run((err, stats): void => {
+      function getResourcePath(
+        webpackModule?: Module | NormalModule | ConcatenatedModule | null
+      ): string | undefined {
+        if (webpackModule && !(webpackModule instanceof ConcatenatedModule)) {
+          return (webpackModule as NormalModule).resource
         }
-        if (closeErr) {
-          return reject(closeErr)
+
+        if (webpackModule?.modules) {
+          // ConcatenatedModule is a collection of modules so we have to go deeper to actually get a path,
+          // at this point we won't know which one so we just grab first module here
+          const [firstSubModule] = webpackModule.modules
+          return getResourcePath(firstSubModule)
+        }
```

Comment:
I actually don't know if we can hit this path or not - I didn't hit it in my testing, but I saw it was used in https://github.com/gatsbyjs/gatsby/blob/0cd90d9bd1f1d93f598fa137341b1438214f25e8/packages/gatsby/src/utils/webpack/plugins/static-query-mapper.ts#L17-L32 so I just added some handling here